### DISCUSSION TITLE
Warn or fail when vcpkg-configuration.json has a double entry between default and the other registries

### DISCUSSION THREAD

**Discussion by Thieum** [microsoft/vcpkg#25386] [Ideas]
When configuring my vcpkg-configuration.json this way:

```
{
    "default-registry": {
        "kind": "git",
        "repository": "http://10.1.102.168/Bonobo.Git.Server/vcpkg-registry.git",
        "baseline": "df99aca6cb95547e0b8af076f04b494fae96fff7"
    },
    "registries": [
      {
        "kind": "git",
        "repository": "http://10.1.102.168/Bonobo.Git.Server/vcpkg-registry.git",
        "baseline": "df99aca6cb95547e0b8af076f04b494fae96fff7",
        "packages": [ "csvhelper", "stringhelper" ]
      },
      {
        "kind": "builtin",
        "baseline": "6cdc26c884ac1c7c4cf2d480b364b773765622a3",
        "packages": [ "gtest", "paho-mqtt" ]
      }
    ]
  }
```

I was able to search for a package, but not install it, without much explanation about why it would fail.

It would be helpful to have a warning or an error in case of misconfiguration.

### Discussed in https://github.com/microsoft/vcpkg/discussions/18852

<div type='discussions-op-text'>

<sup>Originally posted by **Thieum** July  7, 2021</sup>
I certainly made a mistake during the process of adding a new package in my registry. 

I am able to find the package, but I cannot install it.

![image](https://user-images.githubusercontent.com/944894/124783597-30247380-df13-11eb-8fab-dfd0672966b9.png)

There is the git-tree that works for the search:

![image](https://user-images.githubusercontent.com/944894/124784365-db352d00-df13-11eb-9a4b-e02560ceb3e0.png)

But when installing, another git-tree is pulled, which is older and not working:

![image](https://user-images.githubusercontent.com/944894/124784435-edaf6680-df13-11eb-9537-6358d9ae6ee2.png)

My vcpkg-configuration.json is written this way:

```
{
    "default-registry": {
        "kind": "git",
        "repository": "http://10.1.102.168/Bonobo.Git.Server/vcpkg-registry.git",
        "baseline": "df99aca6cb95547e0b8af076f04b494fae96fff7"
    },
    "registries": [
      {
        "kind": "git",
        "repository": "http://10.1.102.168/Bonobo.Git.Server/vcpkg-registry.git",
        "baseline": "df99aca6cb95547e0b8af076f04b494fae96fff7",
        "packages": [ "csvhelper", "stringhelper" ]
      },
      {
        "kind": "builtin",
        "baseline": "6cdc26c884ac1c7c4cf2d480b364b773765622a3",
        "packages": [ "gtest", "paho-mqtt" ]
      }
    ]
  }
```
  
  I would expect search and install to work the same way, at least to pull the resources. I wonder what I'm missing here?

Edit:

using --debug I found that the git commands were using two different commit ids to grab the files - the first one for baseline,json is the good one, the second one for versions is too old: 
![image](https://user-images.githubusercontent.com/944894/124938912-b0f87380-dfd6-11eb-81d7-e9f6475ec86e.png)

Where the second one comes from? What could be the source of this old commit id beside from vcpkg-configuration.json?

Edit:
I managed to bypass this issue by:
 - creating a newer version of the first package I created;
 - installing this newer version locally;
 - installing the other package

It still feels like a hack, and not really sustainable if I have to create new port for every packages everytime I create a new package.

</div>

**Comment by PhoebeHui** [comment]
@ras0219-msft, @vicroms, could you please take a look? I'm not sure where the 2 commits come from?

**Comment by PhoebeHui** [comment]
Ah, @Thieum, so csvhelper is a custom port in your local, we don't have that port in vcpkg now, the error should expect for me.

Could you follow https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/registries.md#adding-a-new-version, and see if it works for you?

**Comment by Thieum** [comment]
@PhoebeHui Yes, those are custom packages in a private registry.

I managed to fix my error, following the answer I got in the mentioned discussion. And It works now.

My issue is more of a feature request, regarding a minimal self-validation around the vcpkg-configuration.json file. In my case, I would have like a warning or an error regarding the fact that I had the same registry as default, and in the subsequent registries list.
